bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2367,7 +2035,324 @@ public String toString () {
 public static Image win32_new(Device device, int type, long handle) {
 	Image image = new Image(device);
 	image.type = type;
-	image.setHandleForZoomLevel(handle, image.getZoom());
+	ImageMetadata imageMetadata = image.new ImageMetadata(handle, image.getZoom());
+	image.setImageMetadataForHandle(imageMetadata, image.getZoom());
 	return image;
 }
+
+class ImageMetadata {
```

Comment:
As we are moving logic into this class now, I would probably renamed it as it is more than data now, something like ImageHandle? And it can be private, can't it?

Reply:
Alright

Reply:
How About ImageHandleProvider, since handle is an interior member of it?

Reply:
I renamed it to `ImageHandle`.